terraform-provider-constellation/internal/provider/cluster_resource.go

```diff
@@ -150,154 +315,593 @@ func (r *ClusterResource) Configure(_ context.Context, req resource.ConfigureReq
 	if req.ProviderData == nil {
 		return
 	}
-
-	// client, ok := req.ProviderData.(*http.Client)
-
-	// if !ok {
-	//	resp.Diagnostics.AddError(
-	//		"Unexpected Resource Configure Type",
-	//		fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
-	//	)
-
-	//	return
-	//}
 }
 
 // Create is called when the resource is created.
 func (r *ClusterResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Read data supplied by Terraform runtime into the model
 	var data ClusterResourceModel
-	// Read Terraform plan data into the model
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Parse and convert values from the Terraform state
+	// to formats the Constellation library can work with.
 
+	att, diags := r.convertAttestationConfig(ctx, data)
+	resp.Diagnostics.Append(diags...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	var tfAttestation attestation
-	diags := data.Attestation.As(ctx, &tfAttestation, basetypes.ObjectAsOptions{})
+	log := &tfContextLogger{ctx: ctx}
+	newDialer := func(validator atls.Validator) *dialer.Dialer {
+		return dialer.New(nil, validator, &net.Dialer{})
+	}
+	r.applier = constellation.NewApplier(log, &nopSpinner{}, newDialer)
```

Comment:
Checking this, but we might need the context of the actual call.

Reply:
Could add additional wrapping to the creation of the applier.
Right now this makes this function non testable since we always use the "real" applier.

Reply:
Obsolete since now moved in the `apply` function

Reply:
Same thing still applies though.
We can't unit test the function if we always use `constellation.NewApplier` in the function under test.
What I was talking about is setting some function for `ClusterResource` on `*ClusterResource.Configure` which returns the Applier interface.